### ISSUE TITLE
Regularization vs Penalty Functions

### ISSUE THREAD

**Issue by SandyaS72** [Statistical-Connectomics-Sp15/intro#108]
Regularization is kind of putting a penalty on having large values for the parameters of your model... so is that considered a penalty function, or am I confusing two completely different things?

**Comment by ajulian3** [comment]
I also found this confusing. Is there an inherent change in mathematical definitions between loss and penalty functions? Regularization minimizes the loss functions, but when large parameter values exist, there is an associated penalty. It is very hard to discern from the definitions.

**Comment by SandyaS72** [comment]
Yes I guess that's another related and slightly confusing point- can't loss functions be considered a "penalty" in a way?

**Comment by aceeccc** [comment]
I thought bias was defined as the minimum loss, in which case I don't think the loss functions are the same as the penalty? And penalty refers to to inefficient model fitting? I'm not completely sure, but that's what I gleaned from class.

**Comment by itsthefedora** [comment]
The best demo of what regularization does is L_1 regularization in regression (*i.e.* LASSO). Straight regression is trying to find a decision rule (weights) that minimizes a risk functional, the expected squared error (read: expected *loss*) under assumptions about the distribution of the involved data:
R = E[L]
LASSO adds a term to the risk functional that makes it *riskier* to have many non-zero parameters in the model, namely
R = E[L] + a |w|_1
where |w|_1 is the L_1 norm of the weights. Since LASSO has a different risk functional than straight regression, it's going to find a different decision rule that minimizes the risk functional -- namely, one that has more zero-valued parameters. This is an explicit way of imposing a bias-variance tradeoff: higher value of a means more zero-valued parameters, which means less variance and more bias.
